File: Models/Definition/DestinyItemSocketEntryPlugItemRandomizedDefinition.cs
namespace bulkybook.Models;

public class DestinyItemSocketEntryPlugItemRandomizedDefinition
{
    public DestinyPlugItemCraftingRequirements craftingRequirements {get;set;}
    public Boolean currentlyCanRoll {get;set;}
    /*  The hash identifier of a DestinyInventoryItemDefinition representing the plug that can be inserted.
Type: uint32
Mapped to Manifest Database Definition: Destiny.Definitions.DestinyInventoryItemDefinition  */
    public UInt32 plugItemHash {get;set;}

}

File: Scripts/Models/DestinyItemSocketEntryPlugItemRandomizedDefinition.ts
import { DestinyPlugItemCraftingRequirements } from "./DestinyPlugItemCraftingRequirements";
export interface DestinyItemSocketEntryPlugItemRandomizedDefinition
{
    craftingRequirements: DestinyPlugItemCraftingRequirements;
    currentlyCanRoll: boolean;
    /*  The hash identifier of a DestinyInventoryItemDefinition representing the plug that can be inserted.
Type: uint32
Mapped to Manifest Database Definition: Destiny.Definitions.DestinyInventoryItemDefinition  */
    plugItemHash: number;
}